crates/sqlbuiltins/src/functions/object_store.rs

```diff
@@ -78,6 +73,19 @@ impl TableFunc for ObjScanTableFunc {
                 "at least one url expected".to_owned(),
             ));
         }
+        let file_compression = urls[0].get_file_compression();
```

Comment:
i was thinking of doing a named argument, but I thought inferring it would be a bit more intuitive for users. 

```sql
select * from csv_scan("path/to/file.csv.gz")
select * from csv_scan("path/to/*.gz")
```

instead of 

```sql
select * from csv_scan("path/to/file.csv.gz", compression => GZIP)
```

> Moreover, we should iterate over all URLs and take the first one to get the file compression type (since the first URL might not have any file extension). Hence, the fallback.

I don't see this as being something we'd want to handle. I think we'd want to either assume they are all the same, or error if they are different. 

```sql
-- this should result in an error. 
select * from csv_scan(["path/to/file.csv", "path/to/file.csv.gz"])
```

Reply:
> i was thinking of doing a named argument, but I thought inferring it would be a bit more intuitive for users.

Yeah, this definitely is more intuitive, but there should be an option to override this. Let's say user has files without extensions (which is not un-common), in that case we do need the option.

So the order should be:
1. See if the user has provided an option -> use it.
2. Check URLs and get the compression -> use it.
3. Assume un-compressed.

> I don't see this as being something we'd want to handle. I think we'd want to either assume they are all the same, or error if they are different.

Again, here the reason is when files are without extensions. Yes we should error in case when the files have different formats, but that shouldn't be the source of truth. We should fallback to extensions if the user doesn't tell us about the compression.

```sql
-- This shouldn't error IMO
select * from csv_scan(['path/to/file', 'path/to/another/file.csv.gz'])
```

So it's safe to iterate over all URLs and break the loop when some file compression is found which is other than "uncompressed" else we just assume uncompressed and move on.

Not too strongly opinionated on this though. But we should **definitely** have named arguments!